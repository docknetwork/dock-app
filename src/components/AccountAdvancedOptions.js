import React from 'react';
import {FormControl, Input, Select, Stack} from 'native-base';
import {InputPopover, SelectToggler} from '../design-system';
import {translate} from '../locales';

export function AccountAdvancedOptions({onChange, form}) {
  return (
    <SelectToggler placeholder={translate('account_advanced_options.advanced_options')}>
      <FormControl>
        <Stack mt={7}>
          <FormControl.Label>
            {translate('account_advanced_options.keypair_type')}
            <InputPopover title={translate('account_advanced_options.keypair_type')}>
              {translate('account_advanced_options.keypair_type_tip')}
            </InputPopover>
          </FormControl.Label>
          <Select
            onValueChange={onChange('keypairType')}
            selectedValue={form.keypairType}>
            <Select.Item
              label={translate('account_advanced_options.keypair_type_sr25519')}
              value="sr25519"
            />
            <Select.Item
              label={translate('account_advanced_options.keypair_type_ed25519')}
              value="ed25519"
            />
            <Select.Item
              label={translate('account_advanced_options.keypair_type_ecdsa')}
              value="ecdsa"
            />
          </Select>
        </Stack>
      </FormControl>
      <FormControl isInvalid={false}>
        <Stack mt={7}>
          <FormControl.Label>
            {translate('account_advanced_options.secret_derivation_path')}
            <InputPopover
              title={translate('account_advanced_options.secret_derivation_path')}>
              {translate('account_advanced_options.secret_derivation_path_tip')}
            </InputPopover>
          </FormControl.Label>
          <Input
            placeholder="//hard/soft//password"
            onChangeText={onChange('derivationPath')}
            autoCapitalize="none"
          />
          <FormControl.ErrorMessage>
            {translate('account_advanced_options.invalid_secret_uri')}
          </FormControl.ErrorMessage>
        </Stack>
      </FormControl>
    </SelectToggler>
  );
}
