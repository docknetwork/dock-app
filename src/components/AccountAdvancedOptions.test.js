import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {AccountAdvancedOptions} from './AccountAdvancedOptions';

const mockStore = configureMockStore();

describe('AccountAdvancedOptions', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <AccountAdvancedOptions
        onChange={jest.fn()}
        form={{
          phrase: 'test',
          _errors: {
            
          }
        }}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
