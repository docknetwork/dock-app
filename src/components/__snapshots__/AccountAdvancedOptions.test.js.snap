// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountAdvancedOptions should render correctly 1`] = `
<Memo(ForwardRef(Box))>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <Memo(ForwardRef(Stack))
      alignItems="center"
      backgroundColor="#27272A"
      borderRadius={6}
      direction="row"
      mt={8}
      pl={5}
      pr={2}
      py="1px"
      width={0.67}
    >
      <Memo(ForwardRef(Box))
        flex={1}
      >
        <Memo(ForwardRef(Text))
          fontSize={14}
          fontWeight={600}
        >
          Advanced options
        </Memo(ForwardRef(Text))>
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        mt={2}
      >
        <Memo(ForwardRef(createdIcon)) />
      </Memo(ForwardRef(Box))>
    </Memo(ForwardRef(Stack))>
  </TouchableWithoutFeedback>
</Memo(ForwardRef(Box))>
`;
